code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/DefaultAuthorizationProvider.java

```diff
@@ -58,36 +54,31 @@ public SchemaFactory getSchemaFactory(Account account, Policy policy, ApiRequest
                 return schemaFactory;
             }
         }
-        List<? extends ProjectMember> projectMembers;
-        if (account != null && account.getKind().equalsIgnoreCase(ProjectConstants.TYPE)) {
-            projectMembers = authDao.getProjectMembersByIdentity(account.getId(), policy.getIdentities());
-            if (projectMembers == null || projectMembers.size() == 0) {
-                return schemaFactories.get(account.getKind());
+        return schemaFactories.get(authDao.getRole(account, policy));
```

Comment:
getSchemaFactory is called from ApiAuthenticator where it iterates through the Authorization providers and returns the first non-null schema factory.

How do you guarantee that we will get the dynamic schema factory?
To ellaborate, let's say you have two authProviders: default and dynamic and your role is project. If defaultAuthProvider is first in the list that is iterated over in ApiAuthenticator, it seems we'd return the default provider without ever looking at the dynamic provider.